ext/ox/gen_load.c

```diff
@@ -193,12 +225,25 @@ nomode_instruct(PInfo pi, const char *target, Attr attrs, const char *content) {
 static void
 add_doctype(PInfo pi, const char *docType) {
     VALUE       n = rb_obj_alloc(ox_doctype_clas);
-    VALUE       s = rb_str_new2(docType);
+    VALUE       s;
+#if HAVE_RB_ENC_INTERNED_STR_CSTR && HAVE_RB_INTERNED_STR_CSTR
+    if (Yes == pi->options->intern_strings) {
+        if (0 != pi->options->rb_enc) {
+            s = rb_enc_interned_str_cstr(docType, pi->options->rb_enc);
+        } else {
+            s = rb_interned_str_cstr(docType);
+        }
+    } else {
+#endif
+    s = rb_str_new2(docType);
 
 #if HAVE_RB_ENC_ASSOCIATE
     if (0 != pi->options->rb_enc) {
         rb_enc_associate(s, pi->options->rb_enc);
     }
+#endif
+#if HAVE_RB_ENC_INTERNED_STR_CSTR && HAVE_RB_INTERNED_STR_CSTR
+    }
```

Comment:
Similar comment to the above here. Don't cross if/else with `#if` conditionals. I know I might have violated that in some places but I have been fixing them as I run into them.

I see the same code duplicated many times. Can this be a put into a function?